packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -177,22 +177,26 @@ exports.onCreateDevServer = (
       type: `application/json`,
     }),
     async (req, res) => {
-      // we are missing handling of node deletion
-      const nodeToUpdate = JSON.parse(JSON.parse(req.body)).data
-
-      await handleWebhookUpdate(
-        {
-          nodeToUpdate,
-          actions,
-          cache,
-          createNodeId,
-          createContentDigest,
-          getNode,
-          reporter,
-          store,
-        },
-        pluginOptions
-      )
+      if (!_.isEmpty(req.body)) {
+        // we are missing handling of node deletion
+        const nodeToUpdate = JSON.parse(JSON.parse(req.body)).data
+
+        await handleWebhookUpdate(
+          {
+            nodeToUpdate,
+            actions,
+            cache,
+            createNodeId,
+            createContentDigest,
+            getNode,
+            reporter,
+            store,
+          },
+          pluginOptions
+        )
+      } else {
+        res.status(500).send(`Received body was empty!`)
```

Comment:
I think it should be code `400`:
> General error when fulfilling the request would cause an invalid state. Domain validation errors, missing data, etc. are some examples.

```suggestion
        res.status(400).send(`Received body was empty!`)
```

Reply:
Yup, thanks, sounds legit, committing : ).